### ISSUE TITLE
`netlify-cli` prevents `got` upgrade to secure version

### ISSUE THREAD

**Issue by ldiqual** [netlify/cli#5331]
### Describe the bug

We got a dependabot alert about the `got` package: https://nvd.nist.gov/vuln/detail/CVE-2022-33987

`yarn why -R got`

```
├─ our-package@workspace:packages/our
│  └─ netlify-cli@npm:12.4.0 (via npm:12.4.0)
│     ├─ @netlify/build@npm:29.1.1 (via npm:^29.1.1)
│     │  ├─ got@npm:10.7.0 (via npm:^10.0.0)
│     │  └─ update-notifier@npm:5.1.0 (via npm:^5.0.0)
│     │     └─ latest-version@npm:5.1.0 (via npm:^5.1.0)
│     │        └─ package-json@npm:6.5.0 (via npm:^6.3.0)
│     │           └─ got@npm:9.6.0 (via npm:^9.6.0)
│     ├─ gh-release-fetch@npm:3.0.2 (via npm:^3.0.0)
│     │  └─ download@npm:8.0.0 (via npm:^8.0.0)
│     │     └─ got@npm:8.3.2 (via npm:^8.3.1)
│     ├─ node-version-alias@npm:1.0.1 (via npm:^1.0.1)
│     │  ├─ all-node-versions@npm:8.0.0 (via npm:^8.0.0)
│     │  │  └─ fetch-node-website@npm:5.0.3 (via npm:^5.0.3)
│     │  │     └─ got@npm:10.7.0 (via npm:^10.7.0)
│     │  └─ normalize-node-version@npm:10.0.0 (via npm:^10.0.0)
│     │     └─ all-node-versions@npm:8.0.0 (via npm:^8.0.0)
│     └─ update-notifier@npm:6.0.2 (via npm:^6.0.0)
│        └─ latest-version@npm:7.0.0 (via npm:^7.0.0)
│           └─ package-json@npm:8.1.0 (via npm:^8.1.0)
│              └─ got@npm:12.5.2 (via npm:^12.1.0)
```

We'd like to resolve this alert by upgrading the `got` package, but this is not possible due to the netlify-cli dependency chain.

### Steps to reproduce

```
yarn add netlify-cli@12.4.0
yarn why -R got
```

### Environment

```
 System:
    OS: macOS 13.0
    CPU: (10) arm64 Apple M1 Pro
    Memory: 90.33 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/Library/Caches/fnm_multishells/96076_1671493299678/bin/node
    Yarn: 3.3.1 - ~/Library/Caches/fnm_multishells/96076_1671493299678/bin/yarn
    npm: 8.19.2 - ~/Library/Caches/fnm_multishells/96076_1671493299678/bin/npm
```

**Comment by danez** [comment]
Thanks for reporting, we are in the process of migrating to ESM and will soon be able to update some dependencies including this one.

**Comment by danez** [comment]
This is now fixed in version 15.0.3
